Make slight tweaks for Debug builds.

RyanWollaeger opened the pull request:
### Description

Testing with a Debug build of a dependent code revealed a comparison of the same constants throws an assertion.

### Changes

+ Modify memcmp usage in == overload comparing physical constants.
+ Update spiner submodule for fix of a recently added assert.

RyanWollaeger committed:
Make slight tweaks for Debug builds.

+ Modify memcmp usage in == overload comparing physical constants.
+ Update spiner submodule for fix of a recently added assert.

RyanWollaeger commented:
@Yurlungur would it be reasonable to convert the bit-wise comparison in the overload to comparing the member data with tolerances (to permit, e.g. 1e-15 differences in units)? Or is that maybe not in the spirit of this comparison?

Yurlungur commented:
> @Yurlungur would it be reasonable to convert the bit-wise comparison in the overload to comparing the member data with tolerances (to permit, e.g. 1e-15 differences in units)? Or is that maybe not in the spirit of this comparison?

I think that would be fine. The current implementation is just laziness on my part I think.

RyanWollaeger commented:
> > @Yurlungur would it be reasonable to convert the bit-wise comparison in the overload to comparing the member data with tolerances (to permit, e.g. 1e-15 differences in units)? Or is that maybe not in the spirit of this comparison?
> 
> I think that would be fine. The current implementation is just laziness on my part I think.

Will make that change quickly here then - thanks!

RyanWollaeger committed:
Use 10xnumeric_limit<Real>::min() in physical constants == overload.

+ Add eps_equal private function to struct (formula from unit tests).
+ Use macro EPS value, similar to opacity classes.

RyanWollaeger commented:
Just made the eps-comparison modification (the memcmp is much more succinct).
